plugins/modules/proxmox_cluster.py

```diff
@@ -0,0 +1,302 @@
+        cluster_name:
+          description: Cluster name from totem.
+          type: str
+          sample: "devcluster"
+        config_version:
+          description: Config version.
+          type: str
+          sample: "1"
+        interface:
+          description: Interface configuration.
+          type: dict
+        ip_version:
+          description: IP version.
+          type: str
+          sample: "ipv4-6"
+        link_mode:
+          description: Link mode.
+          type: str
+          sample: "passive"
+        secauth:
+          description: Whether secure authentication is on.
+          type: str
+          sample: "on"
+        version:
+          description: Totem protocol version.
+          type: str
+          sample: "2"
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    import urllib3
+    from proxmoxer import ProxmoxAPI
+    __MISSING_MODULE__ = False
+except ImportError:
+    __MISSING_MODULE__ = True
+
+if not __MISSING_MODULE__:
+    urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+
+def create_cluster(proxmox, module):
+    cluster_name = module.params['cluster_name'] or module.params['api_host']
+    payload = {'clustername': cluster_name}
+    link0 = module.params.get('link0')
+    link1 = module.params.get('link1')
+
+    if link0:
+        payload['link0'] = link0
+    if link1:
+        payload['link1'] = link1
+
+    if module.check_mode:
+        module.exit_json(changed=True, msg="Cluster '{}' would be created (check mode).".format(cluster_name), cluster=cluster_name)
+
+    try:
+        proxmox.cluster.config.post(**payload)
+    except Exception as e:
+        module.fail_json(msg="Error while creating cluster: {}".format(str(e)))
+
+    module.exit_json(changed=True, msg="Cluster '{}' created.".format(cluster_name), cluster=cluster_name)
+
+
+def get_cluster_info(proxmox, module):
+    try:
+        data = proxmox.cluster.config.join.get()
+    except Exception as e:
+        module.fail_json(msg="Error while obtaining cluster join information: {}".format(str(e)))
+
+    module.exit_json(changed=False, ansible_facts={'cluster_join_info': data})
+
+
+def join_cluster(proxmox, module):
+    master_ip = module.params['master_ip']
+    fingerprint = module.params['fingerprint']
+    api_password = module.params['api_password']
+    cluster_name = module.params.get('cluster_name')
+
+    if not master_ip or not fingerprint:
+        module.fail_json(msg="Action 'join_cluster' requires 'master_ip' and 'fingerprint' arguments.")
+
+    if module.check_mode:
+        module.exit_json(changed=True, msg="Node would join the cluster (check mode).", cluster=cluster_name)
+
+    try:
+        proxmox.cluster.config.join.post(
+            hostname=master_ip,
+            fingerprint=fingerprint,
+            password=api_password
+        )
+
+    except Exception as e:
+        module.fail_json(msg="Error while joining the cluster: {}".format(str(e)))
```

Comment:
may be use a f-string here too.

Reply:
f-Strings failed in the pipeline when running the Py 2.7 tests (see previous commits before squashing).